cli/cli.cpp

```diff
@@ -445,18 +449,19 @@ int main(int argc, char *argv[]) {
 			if (now >= next) {
 				double percent = total == 0 ? 0.0 : (pos * 100.0) / total;
 				double ratio = pos == 0 ? 0.0 : (written * 100.0) / pos;
-				double speed = 0.0;
-				int64_t diff = pos - lastPos;
-				if (diff > 0) {
-					speed = (diff / 1024.0 / 1024.0) * interval_to_s;
-				}
+				History& entry = history[historyPos];
+				int64_t diff = pos - entry.pos;
+				int64_t time = now - entry.time;
+				entry = {pos, now};
+				double speed = time == 0 ? 0 : (diff * b_to_mb) / (time * ns_to_s);
 
 				char temp[128];
 				sprintf(temp, "%3.0f%%, ratio=%3.0f%%, speed=%5.2f MB/s", percent, ratio, speed);
 				statusInfo = temp;
 
 				next = now + interval_ns;
-				lastPos = pos;
+				historyPos++;
+				if (historyPos >= historyLen) { historyPos = 0; }
```

Comment:
nit: Let's avoid single line ifs, and just drop down to the next line.

-[Unknown]